lib/overcommit/configuration.rb

```diff
@@ -37,6 +39,22 @@ def plugin_directory
       File.join(Overcommit::Utils.repo_root, @hash['plugin_directory'] || '.git-hooks')
     end
 
+    # Returns absolute path to directoy for the history of signatures
+    # @return [String] - Absolute path to the signature directory
+    def signature_directory
+      File.join(Overcommit::Utils.repo_root,
+                '.git',
+                @hash['signature_directory'] || 'overcommit-signatures')
+    end
+
+    def signature_history
+      if @hash['signature_directory']
+        @hash['signature_history'].to_i
```

Comment:
Can you elaborate on why setting the signature history is dependent on whether the `signature_directory` configuration option is set? It seems like we should respect `signature_history` regardless.